tests/test_git_import_export.rs

```diff
@@ -76,6 +77,30 @@ fn test_git_import_remote_only_branch() {
     insta::assert_snapshot!(get_branch_output(&test_env, &repo_path), @r###"
     feature1: 9f01a0e04879 message
     "###);
+
+    // Import using git.auto_local_branch = false
+    git_repo
+        .commit(
+            Some("refs/heads/feature2"),
+            &signature,
+            &signature,
+            "message",
+            &tree,
+            &[],
+        )
+        .unwrap();
+    test_env.add_config(
+        br###"
+    [git]
+    auto-local-branch = false
+    "###,
+    );
+    test_env.jj_cmd_success(&repo_path, &["git", "fetch", "--remote=origin"]);
+    insta::assert_snapshot!(get_branch_output(&test_env, &repo_path), @r###"
+    feature1: 9f01a0e04879 message
+    feature2 (deleted)
```

Comment:
Hmm, this means that if you try to push the branch to the remote, you'd end up deleting it there too. I suppose you would typically not do that, and you probably wouldn't even be allowed, so maybe it's not much of a problem. Still, it's a bit ugly. I'll try to keep this case in mind as I think about the other import/export/pull/push problems we've talked about.

Reply:
I the example here, `jj git push --branch feature1` or `jj git push --all` would delete branch from the remote. The local state is considered the desired state.

> With the auto-local-branch semantics (where a missing local branch definitely means it was deleted), this makes sense, but doesn't make sense without auto local branches. It looks like something that could come back to bite us.. hmm..

Right. Do you think this PR is still useful? I do.

Reply:
I agree. I think this PR is still worth keeping for the folks who actually need it. I'll document this sharp edge and we can think about it later.